drivers/serial/Kconfig.stm32

```diff
@@ -7,6 +7,7 @@ config UART_STM32
 	bool "STM32 MCU serial driver"
 	select SERIAL_HAS_DRIVER
 	select SERIAL_SUPPORT_INTERRUPT
+	select SERIAL_SUPPORT_ASYNC
```

Comment:
It seems that CI is failing due to misconfiguration.
Given the trouble, I propose you set this part back to your initial proposal:
```
	select SERIAL_SUPPORT_ASYNC if ($(dt_compat_enabled,$(DT_COMPAT_ST_STM32_DMA_V1)) || $(dt_compat_enabled,$(DT_COMPAT_ST_STM32_DMA_V2)))
	select DMA if ($(dt_compat_enabled,$(DT_COMPAT_ST_STM32_DMA_V1)) || $(dt_compat_enabled,$(DT_COMPAT_ST_STM32_DMA_V2)))
```
And we'll need to find out a better way to handle these dependencies as a latter step.